### PR TITLE
fix powerglove abuse case and stun

### PR DESCRIPTION
**Pull request by zjdtmkhzt**
## About the PR 
Fixes being able to use power gloves via camera
I pulled out the proc used to check this in artifacts into mob, if there is a less hacky way to check this now, please let me know.

I also fixed the power glove disarm function runtiming instead of stunning.
## Why's this needed? 
It's good to fix the bugs/exploits first, so we can properly discuss power glove balance and stuff.

### CODE REVIEW

File: code/mob.dm [hunk 1]
```diff
@@ -3026,3 +3026,7 @@
 		..(0)
 	else
 		..(newdensity)
+
+// to check if someone is abusing cameras with stuff like artifacts, power gloves, etc
+/mob/proc/in_real_view_range(var/turf/T)
+	return src.client && get_dist(T,src) <= (istext(src.client.view) ? 10 : src.client.view)
```
Comment (Tarmunora) [inline code/mob.dm:3032]: use `IN_RANGE` macro perhaps (handles z-levels etc)

**Review by Tarmunora — APPROVED** [review]

File: code/mob.dm (hunk 1 above)
Comment (zjdtmkhzt) [reply code/mob.dm]: Thanks, good idea!

Also special shoutout to byond for requiring parentheses even though there is no logical reason why you would need parentheses there and what the fuck why are you returning the string to me

File: code/mob.dm [hunk 2]
```diff
@@ -3026,3 +3026,7 @@
 		..(0)
 	else
 		..(newdensity)
+
+// to check if someone is abusing cameras with stuff like artifacts, power gloves, etc
+/mob/proc/in_real_view_range(var/turf/T)
+	return src.client && IN_RANGE(T, src, (istext(src.client.view) ? 10 : src.client.view))
```
Comment (ZeWaka) [inline code/mob.dm:3032]: What is 10

Comment (zjdtmkhzt) [reply code/mob.dm:3032]: The default view range that is used for people that somehow don't have clients, I guess.
I could put it into a define, I guess, it's not really used anywhere else though, so I guess I'd put it here?

Comment (ZeWaka) [reply code/mob.dm]: Why isn't this the horizontal screen size?

Comment (zjdtmkhzt) [reply code/mob.dm]: I suppose whoever wrote the original artifact code didn't feel like extracting it from the string or something, I didn't give it that much thought, I guess I could make it less dumb though.

Comment (zjdtmkhzt) [reply code/mob.dm]: I made it get the actual larger measurement of length and width now, it feels a little overengineered, but idk.

Comment (ZeWaka) [reply code/mob.dm]: > I made it get the actual larger measurement of length and width now, it feels a little overengineered, but idk.

It's already a define

Comment (zjdtmkhzt) [reply code/mob.dm]: Ah, I see,, thanks, I will change it to that as soon as I get to it!

Comment (zjdtmkhzt) [reply code/mob.dm]: ok, I just use the widescreen width now, I guess that's fair for everyone

**Pull request merged**